src/features/Chat/components/ActiveWindow/Header/Buttons.tsx

```diff
@@ -20,38 +23,60 @@ type Props = {
 const Buttons = ({ chat, showSearch }: Props) => {
   const { t } = useTranslation('chat');
 
+  const [dialogVariant, setDialogVariant] = useState<DialogVariant>('archive');
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+  const closeDialog = () => setIsDialogOpen(false);
+
+  const openDialog = (variant: DialogVariant) => {
+    setDialogVariant(variant);
+    setIsDialogOpen(true);
+  };
+
   const [updateChatStatus] = useUpdateStatusMutation();
   const userId = useAppSelector(selectUserId);
 
   const updateStatus = (status: ChatFolder) => {
+    if (isDialogOpen) closeDialog();
     if (!userId) return;
     updateChatStatus({ userId, buddyId: chat.buddyId, status });
   };
 
   return (
-    <Container>
-      <IconButton variant="search" sizeInPx={24} onClick={showSearch} />
-      {chat.status === 'ok' ? (
-        <>
-          <StatusButton
-            onClick={() => updateStatus('archived')}
-            icon="archive"
-            text={t('header.archive')}
-          />
-          <StatusButton
-            onClick={() => updateStatus('banned')}
-            icon="block"
-            text={t('header.block')}
-          />
-        </>
-      ) : (
-        <StatusButton
-          onClick={() => updateStatus('ok')}
-          icon="return"
-          text={t('header.restore')}
+    <>
+      {isDialogOpen && (
+        <ConfirmationDialog
+          variant={dialogVariant}
+          buddyName={chat.displayName}
+          close={closeDialog}
+          confirm={() =>
+            updateStatus(dialogVariant === 'archive' ? 'archived' : 'banned')
```

Comment:
I would probably create a map myself for this

```typescript
const nextStatus: Record<DialogVariant, ChatStatus> = {
block: 'banned',
archive: 'archived',
ok: 'ok'
}
```

Then use it just in updateStatus function

```suggestion
            updateStatus(nextStatus[dialogVariant])
```

Reply:
Actually we only need the two because there is no dialog for restoring a conversations.

Reply:
Fixed this!